src/commands/infra/generate.ts

```diff
@@ -175,3 +174,105 @@ export const validateRemoteSource = async (
   }
   return true;
 };
+
+/**
+ * Creates "generated" directory if it does not already exists
+ *
+ * @param projectPath Path to the definition.json file
+ */
+export const createGenerated = async (projectPath: string): Promise<string> => {
+  try {
+    const newGeneratedPath = projectPath + "-generated";
+    mkdirp.sync(newGeneratedPath);
+    logger.info(`Created generated directory: ${newGeneratedPath}`);
+    return newGeneratedPath;
+  } catch (err) {
+    logger.error(`There was a problem creating the generated directory`);
+    logger.error(err);
+    return "";
```

Comment:
@yradsmikham consequences of returning an empty string to downstream methods? I think SPK should fail and have an appropriate error code so that any automation knows there is a filesystem permission/disk space issue.

Reply:
Good idea!